File: src/problems/n206_reverse_linked_list/solution.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	var node *ListNode
	for head != nil {
		node = &ListNode{Val: head.Val, Next: node}
		head = head.Next
	}
	return node
}

File: src/problems/n206_reverse_linked_list/Solution.java
package problems.n206_reverse_linked_list;


class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

public class Solution {

    public ListNode reverseList(ListNode head) {
        ListNode node = null;

        while (head != null) {
            node = new ListNode(head.val, node);
            head = head.next;
        }

        return node;
    }
}
